tests/components/husqvarna_automower/test_init.py

```diff
@@ -127,28 +129,83 @@ async def test_update_failed(
     assert entry.state is entry_state
 
 
+@patch(
+    "homeassistant.components.husqvarna_automower.coordinator.DEFAULT_RECONNECT_TIME", 0
+)
 async def test_websocket_not_available(
     hass: HomeAssistant,
     mock_automower_client: AsyncMock,
     mock_config_entry: MockConfigEntry,
     caplog: pytest.LogCaptureFixture,
     freezer: FrozenDateTimeFactory,
 ) -> None:
-    """Test trying reload the websocket."""
-    mock_automower_client.start_listening.side_effect = HusqvarnaWSServerHandshakeError(
-        "Boom"
-    )
+    """Test trying to reload the websocket."""
+    mock_called = Event()
+    mock_stall = Event()
+    mock_side_effect: Exception | None = None
+
+    async def mock_function():
+        nonlocal mock_side_effect
+        mock_stall.clear()
+        mock_called.set()
+        await mock_stall.wait()
+        if side_effect := mock_side_effect:
+            mock_side_effect = None
+            raise side_effect
+
+    mock_automower_client.auth.websocket_connect.side_effect = mock_function
+
+    # Simulate a WebSocket handshake error
+    mock_side_effect = HusqvarnaWSServerHandshakeError("Boom")
+
+    # Setup integration and verify initial log error message
     await setup_integration(hass, mock_config_entry)
-    assert "Failed to connect to websocket. Trying to reconnect: Boom" in caplog.text
+    await mock_called.wait()
+    mock_called.clear()
     assert mock_automower_client.auth.websocket_connect.call_count == 1
-    assert mock_automower_client.start_listening.call_count == 1
-    assert mock_config_entry.state is ConfigEntryState.LOADED
-    freezer.tick(timedelta(seconds=2))
-    async_fire_time_changed(hass)
+    assert mock_automower_client.start_listening.call_count == 0
+    mock_stall.set()
+    await hass.async_block_till_done()
+    assert "Failed to connect to websocket. Trying to reconnect: Boom" in caplog.text
+
+    # Simulate a successful connection
+    caplog.clear()
+    await mock_called.wait()
+    mock_called.clear()
+    # Prepare for next subtest
+    mock_automower_client.auth.websocket_connect.side_effect = None
+    mock_automower_client.start_listening.side_effect = mock_function
+    # Check outcome
     await hass.async_block_till_done()
     assert mock_automower_client.auth.websocket_connect.call_count == 2
+    assert mock_automower_client.start_listening.call_count == 0
+    assert "Trying to reconnect: Boom" not in caplog.text
+    mock_stall.set()
```

Comment:
Setting the mock_stall event here will raise the same exception for the websocket_connect call as we tested above, even though we have now changed the side effects of the patched functions, since this change was done after.

Reply:
`mock_stall` is never cleared.
```suggestion
```